Update badges workflow is not working

**Describe the bug**
```
Run python_version=$(grep "requires-python" pyproject.toml | grep -o ">=[^\"]*")
  python_version=$(grep "requires-python" pyproject.toml | grep -o ">=[^\"]*")
  python_badge=$(grep -o 'python-[^)]*' README.md)
  new_python_badge="python-$python_version-blue?logo=python"
  sed -i "s/$python_badge/$new_python_badge/g" README.md
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/[3](https://github.com/AllenNeuralDynamics/aind-smartsheet-api/actions/runs/10930916817/job/30344848244#step:5:3).8.18/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.8.18/x6[4](https://github.com/AllenNeuralDynamics/aind-smartsheet-api/actions/runs/10930916817/job/30344848244#step:5:4)/lib
sed: -e expression #1, char 31: unterminated `s' command
Error: Process completed with exit code 1.Run python_version=$(grep "requires-python" pyproject.toml | grep -o ">=[^\"]*")
  python_version=$(grep "requires-python" pyproject.toml | grep -o ">=[^\"]*")
  python_badge=$(grep -o 'python-[^)]*' README.md)
  new_python_badge="python-$python_version-blue?logo=python"
  sed -i "s/$python_badge/$new_python_badge/g" README.md
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/[3](https://github.com/AllenNeuralDynamics/aind-smartsheet-api/actions/runs/10930916817/job/30344848244#step:5:3).8.18/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.8.18/x6[4](https://github.com/AllenNeuralDynamics/aind-smartsheet-api/actions/runs/10930916817/job/30344848244#step:5:4)/lib
sed: -e expression #1, char 31: unterminated `s' command
Error: Process completed with exit code 1.Run python_version=$(grep "requires-python" pyproject.toml | grep -o ">=[^\"]*")
  python_version=$(grep "requires-python" pyproject.toml | grep -o ">=[^\"]*")
  python_badge=$(grep -o 'python-[^)]*' README.md)
  new_python_badge="python-$python_version-blue?logo=python"
  sed -i "s/$python_badge/$new_python_badge/g" README.md
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/[3](https://github.com/AllenNeuralDynamics/aind-smartsheet-api/actions/runs/10930916817/job/30344848244#step:5:3).8.18/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.8.18/x6[4](https://github.com/AllenNeuralDynamics/aind-smartsheet-api/actions/runs/10930916817/job/30344848244#step:5:4)/lib
sed: -e expression #1, char 31: unterminated `s' command
Error: Process completed with exit code 1.
```

**To Reproduce**
Steps to reproduce the behavior:
1. Go to actions
2. Click on tag_and_publish workflow
3. See error

**Expected behavior**
The update_badges workflow should update the README badges correctly.

**Additional context**
Add any other context about the problem here.

--- a/.github/workflows/tag_and_publish.yml
+++ b/.github/workflows/tag_and_publish.yml
@@ -30,7 +30,7 @@ jobs:
     - name: Get interrogate values and Update README.md
       run: |
         interrogate_val=$(interrogate . | grep -o 'actual: [0-9]*\.[0-9]*' | awk '{print $2}')
-        interrogate_badge=$(grep -o 'interrogate-[^)]*' README.md)
+        interrogate_badge=$(grep '^\!\[Interrogate\]' README.md | sed -n 's/.*\(interrogate-[^)]*\).*/\1/p')
         if (( $(echo "$interrogate_val >= 90.00" | bc -l) )); then
             new_interrogate_badge="interrogate-$interrogate_val%25-brightgreen"
         elif (( $(echo "$interrogate_val < 80.00" | bc -l) )); then
@@ -43,7 +43,7 @@ jobs:
       run: |
         coverage run -m unittest discover
         coverage_val=$(coverage report | grep "^TOTAL" | grep -o '[0-9]\+%' | grep -o '[0-9]\+')
-        coverage_badge=$(grep -o "coverage-[^?]*" README.md)
+        coverage_badge=$(grep '^\!\[Coverage\]' README.md | sed -n 's/.*\(coverage-[^)]*\).*/\1/p')
         if (( $(echo "$coverage_val >= 90.00" | bc -l) )); then 
             new_coverage_badge="coverage-$coverage_val%25-brightgreen"
         elif (( $(echo "$coverage_val < 80.00" | bc -l) )); then

--- a/.github/workflows/tag_and_publish.yml
+++ b/.github/workflows/tag_and_publish.yml
@@ -24,7 +24,7 @@ jobs:
     - name: Get Python version and Update README.md
       run: |
         python_version=$(grep "requires-python" pyproject.toml | grep -o ">=[^\"]*")
-        python_badge=$(grep -o 'python-[^)]*' README.md)
+        python_badge=$(grep '^\!\[Python\]' README.md | sed -n 's/.*\(python->[^)]*\).*/\1/p')
         new_python_badge="python-$python_version-blue?logo=python"
         sed -i "s/$python_badge/$new_python_badge/g" README.md
     - name: Get interrogate values and Update README.md

